src/parser/mod.rs

```diff
@@ -6274,6 +6274,30 @@ impl<'a> Parser<'a> {
             self.expect_keyword(Keyword::WITH)?;
             let table_name = self.parse_object_name(false)?;
             AlterTableOperation::SwapWith { table_name }
+        } else if dialect_of!(self is PostgreSqlDialect)
```

Comment:
```suggestion
        } else if dialect_of!(self is PostgreSqlDialect | GenericDialect)
```
if no conflicts we can include support for the generic dialect

Reply:
Done in 25d6a40